### PR TITLE
Allow for different resolutions to be used when running tests

### PR DESCRIPTION
**Pull request by sdclarke**
This pull request will allow for a custom resolution to be passed to os-autoinst when running under Qemu. The variables `XRES` and `YRES` control the resolution, and default to 1024 and 768 respectively.

For now this PR includes a change to allow the use of a video device other than `virtio-gpu-pci` for ARM architectures.

~~More changes are required in order to generate the video at the correct resolution.~~

### CODE REVIEW

**Comment by Martchus** [comment]
The documentation tests fail. Could you document these additional new variables in `doc/backend_vars.asciidoc`?

Otherwise the changes are good. At least the fullstack test should cover the code and catch regressions. If you'd like, you could nevertheless try to extend unit tests. For instance, one could adjust one of the existing invocations in `t/18-backend-qemu.t` or `t/18-qemu-options.t` and check whether the QEMU command-line contains the expected changes.

**Comment by Martchus** [comment]
It works fine using an external video encoder. The internal video encoder also produces images of the right size. However, the Theora video still has the default height (but the correct width). Not sure why that is the case. The code doesn't contain any obvious mistakes.

File: backend/baseclass.pm [hunk 1]
```diff
@@ -315,6 +315,9 @@ sub start_encoder ($self) {
     my $cwd = Cwd::getcwd;
     my @cmd = (qw(nice -n 19), "$bmwqemu::scriptdir/videoencoder", "$cwd/video.ogv");
     push(@cmd, '-n') if $bmwqemu::vars{NOVIDEO} || ($has_external_video_encoder_configured && !$bmwqemu::vars{EXTERNAL_VIDEO_ENCODER_ADDITIONALLY});
+    my $xres = $bmwqemu::vars{XRES} // '1024';
```
Comment (okurz) [inline backend/baseclass.pm:318]: Even if it's just the default value now we have three places with the resolution default values set, here as a string, later as integer. This is also likely the reason why the variable documentation test complains about needing it for both "common" as well as "qemu" but we should also not duplicate in the documentation. As we need the resolutions already here and not only for qemu I would say make it a property of the class itself in https://github.com/os-autoinst/os-autoinst/pull/2070/files#diff-ee84fe6541d1e2e971e9e0d4bf0187272df8fa4ac878d033b1c8c600325e9f6dR61 and set the default there and use the same variable within backend/qemu.pm . As "backend::qemu" *is* a "backend:baseclass" you just have the variable available there directly as well.

**Review by okurz — CHANGES_REQUESTED** [review]

File: backend/baseclass.pm (hunk 1 above)
Comment (sdclarke) [reply backend/baseclass.pm]: I think I've managed to do what you've suggested here in the latest commit.

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]
I like the change, also the video device. Fewer magic numbers and nicely documented!

File: backend/baseclass.pm [hunk 2]
```diff
@@ -315,6 +317,7 @@ sub start_encoder ($self) {
     my $cwd = Cwd::getcwd;
     my @cmd = (qw(nice -n 19), "$bmwqemu::scriptdir/videoencoder", "$cwd/video.ogv");
     push(@cmd, '-n') if $bmwqemu::vars{NOVIDEO} || ($has_external_video_encoder_configured && !$bmwqemu::vars{EXTERNAL_VIDEO_ENCODER_ADDITIONALLY});
+    push(@cmd, "-x $self->{xres} -y $self->{yres}");
```
[the remark this answers is not in the document]
Comment (Martchus) [reply backend/baseclass.pm:320]: > I'm wondering why this works.

Good catch - I suppose that's the reason why it does *not* work. (See https://github.com/os-autoinst/os-autoinst/pull/2070#issuecomment-1136087559= and my next comment for what's not working.)

[the remark this answers is not in the document]
Comment (perlpunk) [reply backend/baseclass.pm:320]: Ok, I confirmed: if I call `./build/videoencoder "-x 1 -y 2"` then it receives `xres` but not `yres`.
It never complains about the rest of the option and just takes the number, it seems.
Will make a PR.
